File: fitness_app/lib/models/conversation.dart
import 'message.dart';

class Conversation {
  final String id;
  final String title;
  final DateTime createdAt;
  final DateTime lastUpdated;
  final List<Message>? messages;

  Conversation({
    required this.id,
    required this.title,
    required this.createdAt,
    required this.lastUpdated,
    this.messages,
  });

  factory Conversation.fromJson(Map<String, dynamic> json) {
    return Conversation(
      id: json['id'],
      title: json['title'],
      createdAt: DateTime.parse(json['createdAt']),
      lastUpdated: DateTime.parse(json['lastUpdated']),
      messages: json['messages'] != null
          ? (json['messages'] as List)
              .map((m) => Message.fromJson(m))
              .toList()
          : null, // ✅ Safe fallback
    );
  }
}

File: src/dotnet/models/Conversation.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using dotnet.DAL;

namespace dotnet.Models
{
    public class Conversation
    {
        [Key]
        public Guid Id { get; set; }
        [Required]
        public int UserId { get; set; }
        [ForeignKey("UserId")]
        public UserDAL User { get; set; } = null!;
        public string? Title { get; set; }
        [Required]
        public DateTime CreatedAt { get; set; }
        [Required]
        public DateTime LastUpdated { get; set; }
        public ICollection<Message> Messages { get; set; } = new List<Message>();
        public string? SystemContext { get; set; }
    }
}